### PR TITLE
Fix wrong dependencies for on_success(), on_failure(), on_error() functions of Task

### PR DESCRIPTION
**Pull request by xiang9156**
While I was using on_success, on_failure and on_error functions inside Task, I found that the dependencies looked incorrect.
According to the description of the functions, the correct logic should be like this:
```
If A.on_success(B), it means that A >> B and execute B when A.status == success
if A.on_failure(B), it means that A >> B and execute B when A.status == failure
If A.on_error(B), it means that A >> B and execute B when A.status == error
```
However, according to my understanding of the existing code, the logic looks like this:
```
If A.on_success(B), B >> A and execute A when B.status == success
If A.on_failure(B),  B >> A and execute A when B.status == failure
If A.on_error(B), B >> A and execute A when B.status == error
```
@flaviuvadan  Correct me if I'm wrong, hopefully this PR is the correct fix. :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,12 @@ The general format is:
 
 ```
 
+# 2.9.0 - DATE (04/08/2022)
```
Comment (flaviuvadan) [inline CHANGELOG.md:32]: Can this be a PATCH since it's a bug fix? 🙂

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-2.8.0
+2.9.0
```
Comment (flaviuvadan) [inline VERSION:1]:
```suggestion
2.8.1
```

**Comment by flaviuvadan** [comment]
Thank you so much for catching this @xiang9156! 🚀  It was recently reported in https://github.com/argoproj-labs/hera-workflows/issues/128 by @RunningIkkyu 🙂 I left a minor request for the version!

File: VERSION (hunk 2 above)
Comment (xiang9156) [reply VERSION]: updated :)

File: CHANGELOG.md (hunk 1 above)
Comment (xiang9156) [reply CHANGELOG.md]: Sure thing! Thanks for the comment :)

Comment (xiang9156) [reply CHANGELOG.md]: done

**Pull request merged**